### PR TITLE
Implement checkout integration mode

### PR DESCRIPTION
**Pull request by goncafa**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -33,7 +33,7 @@
     "comma-spacing": [2, { "before": false, "after": true }],
     "comma-style": [2, "last"],
     "complexity": 0,
-    "consistent-return": 2,
+    // "consistent-return": 2,
```
Comment (leosoto) [inline .eslintrc]: Why are we commenting out all these rules? If they are a bad idea, we should remove them. If they are a good idea and only sometimes we need to break them, then we can add comments to the infringing code to disable the offending eslint rule there.

File: README.md [hunk 2]
```diff
@@ -39,22 +39,48 @@ Agrega el siguiente HTML justo antes de cerrar tu etiqueta body:
...
         });
 </script>
 ```
 ## Modo de uso
 
+Existen dos formas de integrar el pago Onepay en tu comercio:
+1. QR Directo
+2. Checkout
+
+En nuestra modalidad QR Directo tienes la libertad de poder pintar el código QR donde tu decidas dentro tu pagina web,
```
Comment (leosoto) [inline README.md]: Yo diría "dibujar" en lugar de "pintar"

File: README.md [hunk 3]
```diff
@@ -39,22 +39,48 @@ Agrega el siguiente HTML justo antes de cerrar tu etiqueta body:
...
+mientras que con la modalidad de Checkout se desplegara un Modal que contendrá el QR y toda la funcionalidad necesaria
```
Comment (leosoto) [inline README.md]: Falta tilde en desplegará

File: README.md [hunk 4]
```diff
@@ -39,22 +39,48 @@ Agrega el siguiente HTML justo antes de cerrar tu etiqueta body:
...
 
+Existen dos formas de integrar el pago Onepay en tu comercio:
+1. QR Directo
+2. Checkout
+
+En nuestra modalidad QR Directo tienes la libertad de poder pintar el código QR donde tu decidas dentro tu pagina web,
+mientras que con la modalidad de Checkout se desplegara un Modal que contendrá el QR y toda la funcionalidad necesaria 
+para completar el pago en forma satisfactoria.
+
+En caso que decidas integrar bajo la modalidad de QR Directo eres tu el encargado de implementar que pasara en tu pagina
```
Comment (leosoto) [inline README.md]: Falta tilde en tú

Comment (leosoto) [reply README.md]: Y también en pasará y en página

File: README.md [hunk 5]
```diff
@@ -89,22 +115,27 @@ var transaction = {
 };
 ```
 
-Los valores de `occ`, `ott`, `externalUniqueNumber` y `qrCodeAsBase64` deben ser obtenidas en tu backend al crear una transacción Onepay y transmitidas a tu frontend.
+Los valores de `occ`, `ott`, `externalUniqueNumber` y `qrCodeAsBase64` deben ser obtenidas en tu backend al crear una
```
Comment (leosoto) [inline README.md]: Debiera decir obtenidos en lugar de obtenidas (el sujeto es "valores", que tiene género masculino)

File: README.md [hunk 6]
```diff
@@ -129,17 +161,48 @@ Lo anterior es importante ya que debemos indicarle luego al SDK o librería el I
...
-## Ahora le toca al usuario
+## Integración Checkout
+### Crear requerimiento
+
+Lo primero que debes crear es el objeto de requerimiento para el SDK el cual se arma de la siguiente forma:
 
-De aquí en adelante es el usuario quien comenzará a interactuar con la aplicación móvil que escaneará el código QR. Luego nuestra pagina se irá enterando de los cambios de estados cuando el SDK invoque a los diferentes callbacks que has implementado para poder personalizar la experiencia de tus clientes.
+````javascript 1.5
+var options = {
+  endpoint: './transaction-create.html',
```
Comment (leosoto) [inline README.md]: Saquemos esa extensión html que no aporta mucho

File: README.md [hunk 7]
```diff
@@ -129,17 +161,48 @@ Lo anterior es importante ya que debemos indicarle luego al SDK o librería el I
...
+  commerceLogo: '/onepay-sdk-example/images/icons/logo-01.png',
+  callbackUrl: './transaction-commit.html'
```
Comment (leosoto) [inline README.md]: Saquemos esa extensión html de acá también

File: README.md [hunk 8]
```diff
@@ -129,17 +161,48 @@ Lo anterior es importante ya que debemos indicarle luego al SDK o librería el I
...
+};
+````
+
+1. `endpoint` : corresponse a la URL que tiene la lógica de crear la transacción usando alguno de nuestros SDK
```
Comment (leosoto) [inline README.md]: Y cómo adivino que cosa debe retornar ese endpoint y en qué formato?

File: README.md [hunk 9]
```diff
@@ -129,17 +161,48 @@ Lo anterior es importante ya que debemos indicarle luego al SDK o librería el I
...
+  commerceLogo: '/onepay-sdk-example/images/icons/logo-01.png',
+  callbackUrl: './transaction-commit.html'
+};
+````
+
+1. `endpoint` : corresponse a la URL que tiene la lógica de crear la transacción usando alguno de nuestros SDK 
+disponibles para backend o invocando directamente al API de Onepay.
+2. `commerceLogo` : El el logo de comercio que se mostrara eb el modal. Se debe usar el path directo a la imagen o bien
+la URL de la misma
+3. `callbackUrl` : URL que invocara desde el SDK una vez que la transacción ha sido autorizada por el comercio. En este
```
Comment (leosoto) [inline README.md]: Lo mismo acá: ¿Qué parámetros recibe este callback?

File: README.md [hunk 10]
```diff
@@ -129,17 +161,48 @@ Lo anterior es importante ya que debemos indicarle luego al SDK o librería el I
...
+callback el comercio debe hacer el confirmación de la transacción, para lo cual dispone de 30 segundos desde que la
+transacción se autorizo, de lo contrario esta sera automáticamente reversada.
+
```
Comment (leosoto) [inline README.md]: ¿...y qué pasa si falla el pago?

File: README.md [hunk 11]
```diff
@@ -39,22 +39,48 @@ Agrega el siguiente HTML justo antes de cerrar tu etiqueta body:
...
+    })(false, document, "https://cdn.rawgit.com/TransbankDevelopers/transbank-sdk-js-onepay/v1.2.0/lib/onepay.min.js", 
+        "script",window, function () {
             console.log("Onepay JS library successfully loaded.");
         });
 </script>
 ```
 ## Modo de uso
 
+Existen dos formas de integrar el pago Onepay en tu comercio:
+1. QR Directo
```
Comment (leosoto) [inline README.md]: Invirtamos el orden. Pongamos checkout primero (que es la integración fácil y rápida) y QR directo segundo (que es el "hágalo ud mismo")

File: .eslintrc (hunk 1 above)
Comment (goncafa) [reply .eslintrc]: I will remove those I think are not good rules

File: README.md (hunk 2 above)
Comment (goncafa) [reply README.md]: Ok, hare el cambio.

File: README.md (hunk 3 above)
Comment (goncafa) [reply README.md]: Gracias, haré la corrección

File: README.md (hunk 5 above)
Comment (goncafa) [reply README.md]: Tienes razón. Haré el cambio

File: README.md (hunk 6 above)
Comment (goncafa) [reply README.md]: Ok, eliminare todos los .html

File: README.md (hunk 8 above)
Comment (goncafa) [reply README.md]: Lo corregiré

File: README.md (hunk 9 above)
Comment (goncafa) [reply README.md]: Lo corregiré

File: README.md (hunk 10 above)
Comment (goncafa) [reply README.md]: Pondré un texto que diga que sucede en tal caso.

File: src/httputil.js [hunk 12]
```diff
@@ -0,0 +1,37 @@
+class HttpUtil {
+  constructor() {}
+
+  getHttpRequest() {
+    if (window.XMLHttpRequest) {return new XMLHttpRequest();}
```
Comment (eugenioLopezRamos) [inline src/httputil.js]: Esto no se iba a eliminar?

File: src/onepay-direct-qr.js [hunk 13]
```diff
@@ -1,9 +1,16 @@
...
 
-class Onepay {
+class OnepayDirectQr {
   constructor(transaction) {
-    this.version = '1.1.0';
-    this.transaction = transaction;
+    this.version = '1.2.0';
+
+    if (transaction) {
+      this.transaction = transaction;
```
Comment (eugenioLopezRamos) [inline src/onepay-direct-qr.js]: I don't understand why it's needed to do this, if transaction is not given then transaction will === undefined/null, which is basically the same result as not doing that ```if (transaction)```block

File: src/onepay-direct-qr.js [hunk 14]
```diff
@@ -16,23 +23,17 @@ class Onepay {
...
-      let socket = new OnepayWebSocket(instance.transaction);
+    this.qrCodeAsBase64 = this.transaction.qrCodeAsBase64;
 
-      socket.connect(() => resolve());
-    });
+    let socket = new OnepayWebSocket(this.transaction);
 
-    socketSubscribePromise.then(function () {
+    const instance = this;
+    socket.connect(function () {
```
Comment (eugenioLopezRamos) [inline src/onepay-direct-qr.js]: I think you can remove the ```const instance = this; ``` line and use ```this.qrCodeAsBase64``` below if you change the function() {} callback into an () => {} callback

File: src/onepay.js [hunk 15]
```diff
@@ -0,0 +1,67 @@
+const OnepayDirectQr = require('./onepay-direct-qr');
+const OnepayCheckout = require('./onepay-checkout');
+const SmartPhone = require('./smartphone');
+
+class Onepay {
+  constructor(transaction) {
+    if (transaction) {
+      this.transaction = transaction;
```
Comment (eugenioLopezRamos) [inline src/onepay.js]: Same comment as the OnepayDirectQr constructor

File: src/onepay.js [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+  }
+}
+
+SmartPhone.userAgent = null;
+
+if (typeof window === 'function' || typeof window === 'object') {
+  SmartPhone.setUserAgent(navigator.userAgent);
+}
+
+if (typeof exports !== 'undefined') {
```
Comment (eugenioLopezRamos) [inline src/onepay.js]: It seems like this is for checking if the code is being used on node.js (https://stackoverflow.com/questions/12622895/what-is-exports-object-for)

File: src/onepay.js [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+
+        next();
+      };
+    }
+
+    return SmartPhone;
+
+  };
+
+  if (typeof module !== 'undefined' && module.exports) {
```
Comment (eugenioLopezRamos) [inline src/onepay.js]: Same, I think.

File: src/httputil.js (hunk 12 above)
Comment (goncafa) [reply src/httputil.js]: You right, i dont know really why it is there again... propably becouse the checkout use it in the original implement but i have removed this dependence so I will remove it again.

File: src/onepay.js (hunk 17 above)
Comment (goncafa) [reply src/onepay.js]: same reply of the another same question.

File: README.md (hunk 11 above)
Comment (goncafa) [reply README.md]: Yes, i will fix it in the new push

File: src/onepay-direct-qr.js (hunk 14 above)
Comment (goncafa) [reply src/onepay-direct-qr.js]: I will make a test and if it works i will change it.

Comment (goncafa) [reply src/onepay-direct-qr.js]: It works :)

File: src/onepay-direct-qr.js (hunk 13 above)
Comment (eugenioLopezRamos) [reply src/onepay-direct-qr.js]: What I mean is, if someone calls ```let directQr = new OnepayDirectQr();``` (no parameter given) then ```directQr.transactiom``` will be `undefined`, which is the same that would happen if you didn't do that ```if``` (unless someone called ```new OnepayDirectQr(false|null)```

File: src/onepay-direct-qr.js (hunk 14 above)
Comment (eugenioLopezRamos) [reply src/onepay-direct-qr.js]: Great!

File: src/onepay-direct-qr.js (hunk 13 above)
Comment (goncafa) [reply src/onepay-direct-qr.js]: I think you are right so i will remove the if

File: src/onepay.js (hunk 16 above)
Comment (goncafa) [reply src/onepay.js]: @leosoto @RodrigoAyala @afiebig please your opinions about this wil be helpfull in order to merge this PR asap.

**Comment by goncafa** [comment]
I have made all corrections @leosoto @eugenioLopezRamos

**Pull request merged**